docs/failsafe-controller.md

```diff
@@ -0,0 +1,192 @@
+
+Syntax sugar is provided for Spock users to make the mocking look more like Spock's mocking. For example, the above could be re-written as follow:
+```
+  def pingController = new FailsafeController('SolrClient Ping') >>> [ 
+    doNothing(), 
+    doThrow(pingError) + doThrow(new SolrException(ErrorCode.UNKNOWN, 'failed').untilNotifiedTo('connect') + doNothing()
+  ]
+```
+or again as:
+```
+  def pingController = new FailsafeController('SolrClient Ping') >> doNothing() >> doThrow(pingError) + doThrow(new SolrException(ErrorCode.UNKNOWN, 'failed').untilNotifiedTo('connect') + doNothing(
+```
+### Conditions
+The Failsafe controller maintains a simple set of conditions that have been notified since it was created. Each condition is simply identified using a unique name. As explained before, it provides a very simple latching mechanism between Failsafe's execution attempts and the main test logic.
+
+Conditions can be notified via the _Notify_ action when responding to Failsafe attempts or via the `net.jodah.failsafe.FailsafeController.notify(java.lang.String)` or `net.jodah.failsafe.FailsafeController.notifyTo(java.lang.String)` methods on the controller by the test code. To wait on a condition, simply use the _Wait_ action, customized an action with _Until Notified_ or again call one of the two `net.jodah.failsafe.FailsafeController.waitFor(java.lang.String)` or `net.jodah.failsafe.FailsafeController.waitTo(java.lang.String)` methods on the controller.
+
+### Verification
+Once a test is completed, the controllers in play should be verified to see if any errors occurred or again if all recorded expected executions and actions have been processed. This can be done via the `net.jodah.failsafe.FailsafeController.verify()` method.
+
+### Timing, delays, and blockages
+To simplify the design of the controller all waits that are designed to block execution until something happens do not expect a timeout to be specified. This is primarily due to the fact that time should be removed from the equation when we are performing asynchronous testing such that the tests ends up being deterministic. This is also why the Failsafe controller automatically compress time for the specified retry policy. Because of this and because the code under test could also be buggy and not behave as expected, the test might actually block. It is therefore the responsibility of the test writer to take advantage of the test framework capability to associate a timeout to a given test such that when a condition like this occurs, the test is aborted and cleaned up which is where the Failsafe controller could be shutdown to unblock all places where it might be waiting for something to happen.
```

Comment:
s/b "...that the tests end up being..."
s/b "...automatically compresses time..."